README.md

```diff
@@ -209,16 +176,26 @@ and improve:
 ![Architecture](docs/img/modin_architecture.png)
 
 Visit the [Documentation](https://modin.readthedocs.io/en/latest/development/architecture.html) for
```

Comment:
I thought this was referring to the overall readthedocs, which is repetitive, but its actually the architecture section. I think we can remove this altogether and just link at the top by saying: 

```
We designed [Modin's architecture](https://modin.readthedocs.io/en/latest/development/architecture.html) to be modular so we can plug in different components as they develop and improve:
```